activesupport/lib/active_support/lazy_load_hooks.rb

```diff
@@ -27,19 +27,34 @@ def self.extended(base) # :nodoc:
       base.class_eval do
         @load_hooks = Hash.new { |h, k| h[k] = [] }
         @loaded     = Hash.new { |h, k| h[k] = [] }
+        @only_once  = Hash.new { |h, k| h[k] = [] }
```

Comment:
I wonder if we can delete the rook from `@load_hooks` just after it is executed if `:only_once` is true instead of adding a new hash.